Content.Server/PowerSink/PowerSinkSystem.cs

```diff
@@ -2,12 +2,32 @@
 using Content.Server.Power.Components;
 using Content.Shared.Examine;
 using Robust.Shared.Utility;
+using Content.Server.Chat.Systems;
+using Content.Server.Station.Systems;
+using Robust.Shared.Timing;
 
 namespace Content.Server.PowerSink
 {
     public sealed class PowerSinkSystem : EntitySystem
     {
+        /// <summary>
+        /// Percentage of battery full to trigger the announcement warning at.
+        /// </summary>
+        private const float WarningMessageThreshold = 0.85f;
+
+        private readonly float[] WarningSoundThresholds = new float[] { .90f, .95f, .98f };
+
+        /// <summary>
+        /// Length of time to delay explosion from battery full state -- this is used to play
+        /// a brief SFX winding up the explosion.
+        /// </summary>
+        /// <returns></returns>
+        private readonly TimeSpan ExplosionDelayTime = new TimeSpan(0, 0, 0, 1, 465);
+
         [Dependency] private readonly ExplosionSystem _explosionSystem = default!;
+        [Dependency] private readonly IEntitySystemManager _entitySystemManager = default!;
```

Comment:
Just use systemdependencies instead, plus EntityManager already has a ref to IEntitySystemManager (accessible via EntitySysManager iirc).

Reply:
Please explain what systemdependencies is and the benefits of using it

Reply:
Put [Dependency] private readonly XYZSystem instead of using Get<XYZSystem> every time.

Reply:
Ah, I see, so this isn't requesting a change for this code block, this is just notating an example to be referred to from the other code change request further down.  Got it